analyzer/runtime/incoming.go

```diff
@@ -116,11 +143,12 @@ func extractAddressPreimage(as *sdkTypes.AddressSpec) (*AddressPreimageData, err
 	}, nil
 }
 
-func registerAddressSpec(addressPreimages map[string]*AddressPreimageData, as *sdkTypes.AddressSpec) (string, error) {
-	addr, err := common.StringifyAddressSpec(as)
+func registerAddressSpec(addressPreimages map[apiTypes.Address]*AddressPreimageData, as *sdkTypes.AddressSpec) (apiTypes.Address, error) {
+	addrStr, err := common.StringifyAddressSpec(as)
 	if err != nil {
 		return "", err
 	}
+	addr := apiTypes.Address(addrStr)
```

Comment:
is there much of this casting? we should modify all of uncategorized/addresses.go to use apiTypes.Address. this PR or elsewhere

Reply:
There was a bit more than I'd like yeah. updated the address fns in uncategorized/addresses.go

Reply:
thanks!